### ISSUE TITLE
Very high `Gradle` sync times on `Idea 2021.1`

### ISSUE THREAD

**Issue by gmazzo** [serenity-bdd/serenity-gradle-plugin#1]
Hi there, this plugins seems to be some kind of broken with the latest version of `IntellJ Idea 2021.1`. Terminal build seems not to be affected.

We have a big backend project with many modules, and if we switch to latest version of the IDE the project's sync becomes pretty much hang (actually it takes about 30 minutes to finish).

### Research
After debugging `Gradle`’s deamon, I find out the problem was this `net.serenity-bdd.aggregator` which is doing a pretty strange logic by scanning and changing `Gradle`’s `Configuration` paths:
https://github.com/serenity-bdd/serenity-gradle-plugin/blob/main/src/main/groovy/net/serenitybdd/plugins/gradle/SerenityPlugin.groovy#L176

That makes the sync process to enter in a loop on Idea 2021.
![image](https://user-images.githubusercontent.com/513566/114713505-7b265480-9d31-11eb-98ee-a9dfb63111bc.png)

After removing that plugin, sync times went back into a decent time: `9secs`
![image](https://user-images.githubusercontent.com/513566/114713611-985b2300-9d31-11eb-97ab-23d9c651129f.png)

**Comment by wakaleo** [comment]
I don't think the Gradle code has changed for quite a while; feel free to have a play and see if you can find anything to propose in a PR though.

**Comment by wakaleo** [comment]
That code doesn't seem to do anything related to the Gradle configuration - what makes you think there is a loop involved?

**Comment by gmazzo** [comment]
> That code doesn't seem to do anything related to the Gradle configuration - what makes you think there is a loop involved?

Probably it's not. I didn't go deeper on the plugin's logic to be honest. I've just paused the daemon's JVM after 10 minutes of "running" and I've found it was at that specific stacktrace I've shared in the first screenshot.

It'd be nice if you can take a look at it, probably we won't be the only ones having issues as latest Idea is rolling out.

**Comment by gmazzo** [comment]
This may give you a clue, it seems it's scanning my whole disk for `serenity.conf` files 🤔 
![image](https://user-images.githubusercontent.com/513566/114882596-6666c100-9e04-11eb-9be4-a453b072adba.png)

**Comment by wakaleo** [comment]
Hmmm, yeah, that could be an issue. I think there was a Gradle bug related to the user.dir variable (it should be the user's home directory, not the root directory). That said, it just checks for the presence of a serenity.conf file there, so there shouldn't be a loop I think.

**Comment by gmazzo** [comment]
> Hmmm, yeah, that could be an issue. I think there was a Gradle bug related to the user.dir variable (it should be the user's home directory, not the root directory). That said, it just checks for the presence of a serenity.conf file there, so there shouldn't be a loop I think.

Yes, even if you just look for a file, scanning the whole disk it's an expensive operation. That's why it was taking 30' to finish for us.

Do you think you can do any workaround for this? Shouldn't this "scanning" be done at EXECUTION time instead of CONFIGURATION time?

**Comment by wakaleo** [comment]
As far as I know, it doesn't scan, it merely checks for the presence of the file on 4 locations.

**Comment by gmazzo** [comment]
> As far as I know, it doesn't scan, it merely checks for the presence of the file in 3 locations.

It is, look at the stackstrace here:
![image](https://user-images.githubusercontent.com/513566/115420788-b7f3bf00-a1fb-11eb-856d-4f18b6d91e2e.png)
The `serenityConfFileInASensibleLocation` of `PropertiesFileLocalPreferences` is calling `SearchForFilesWithName.matching(Paths.get(".")...`.

As `user.dir` (current directory) is returning `/`, `Paths.get(".")` will be `/` also. That will traverse the whole disk.

**Comment by ricardorlg-aval** [comment]
I have the same problem, It never finishes configuring a project in Intellij, in the terminal everything works, Also I just try with an old version of the plugin,  2.1.0 and there is no problem, @wakaleo has u seen this problem in Intellij last version?
